lib/src/main/java/com/cloudbees/groovy/cps/impl/FunctionCallBlock.java

```diff
@@ -0,0 +1,141 @@
+package com.cloudbees.groovy.cps.impl;
+
+import com.cloudbees.groovy.cps.Block;
+import com.cloudbees.groovy.cps.Continuable;
+import com.cloudbees.groovy.cps.Continuation;
+import com.cloudbees.groovy.cps.Env;
+import com.cloudbees.groovy.cps.Next;
+import com.cloudbees.groovy.cps.sandbox.CallSiteTag;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import static com.cloudbees.groovy.cps.impl.SourceLocation.UNKNOWN;
+
+/**
+ * lhs.name(arg,arg,...)
+ *
+ * @author Kohsuke Kawaguchi
+ */
+public class FunctionCallBlock extends CallSiteBlockSupport {
+    /**
+     * Receiver of the call
+     */
+    private final Block lhsExp;
+
+    /**
+     * Method name.
+     * {@code "<init>"} to call constructor
+     */
+    private final Block nameExp;
+
+    /**
+     * Arguments to the call.
+     */
+    private final Block[] argExps;
+
+    private final SourceLocation loc;
+
+    private final boolean safe;
+
+    @SuppressFBWarnings(value = "EI_EXPOSE_REP2", justification = "argExps is not modified by existing callers after being passed to this method and we want to avoid cloning every array")
```

Comment:
SpotBugs didn't like that we are directly storing the mutable `argExps` array passed in as a parameter. This code is only used internally, and nothing ever tries to mutate `argExps`, so I think it is fine to suppress the warning, especially since we don't really want to increase the overhead of this code.